### PR TITLE
Adding SPM support

### LINKED ISSUE #139: Request for Swift package manager support
Good day!
Please, add support for Swift Package Manager. 
Thank you!

### PR DESCRIPTION
**Pull request by bsarrazin**
# iOS SDK

## What did you accomplish?

- Added support for Swift Package Manager
- Fixes #139

## How do we test the changes introduced in this PR?

Create a project that consumes Split through SPM, or another package that has Split as a dependency.

## Extra Notes

- I have everything figured out and working, except for the module map problem in the Xcode project itself. I need some help with this if you have some experience with this.

image

### CODE REVIEW

**Comment by bsarrazin** [comment]
## Extra Extra Notes

I've noticed that the CI pipeline doesn't in fact run the unit tests, so the config for Carthage + Swifter isn't being used at all. Because of this, the unit tests aren't compiling and aren't even run on CI.
The xcode project isn't actually necessary, this whole project can be fully managed without an `.xcodeproj` file and can only use the `Package.swift` file.

Is there interest in moving this repo towards an SPM only project? (this doesn't impact CocoaPods / Carthage support)

File: Split.podspec [hunk 1]
```diff
@@ -4,16 +4,24 @@ Pod::Spec.new do |s|
...
+    'Fernando Martin' => 'fernando@split.io'
+  }
+  s.source = { git: 'https://github.com/splitio/ios-client.git', tag: s.version.to_s }
   s.ios.deployment_target = '9.0'
   s.frameworks = 'Foundation'
   s.swift_versions = ['4.0', '4.2', '5.0', '5.1', '5.2', '5.3']
-  s.source_files = 'Split/**/*.{swift}','Split/Common/Utils/JFBCrypt/*.{h,m}'
+  s.source_files = [
+    'Split/**/*.{swift}',
+    'Split/Common/Utils/JFBCrypt/**/*.{h,m}'
```
Comment (bsarrazin) [inline Split.podspec:24]: ℹ️ aside from a bit of cleanup, this is the only meaningful change, it makes the folder recursive to fetch source files from subfolders

File: Split.xcodeproj/JFBCrypt_Info.plist [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bsarrazin) [inline Split.xcodeproj/JFBCrypt_Info.plist:1]: ℹ️ Generated by `swift package generate-xcodeproj`

File: Split.xcodeproj/Split_Info.plist [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bsarrazin) [inline Split.xcodeproj/Split_Info.plist:1]: ℹ️ Generated by `swift package generate-xcodeproj`

File: Split.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -1,2682 +1,3324 @@
 // !$*UTF8*$!
```
Comment (bsarrazin) [inline Split.xcodeproj/project.pbxproj:1]: ℹ️ Generated by `swift package generate-xcodeproj`

File: Split.xcodeproj/project.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bsarrazin) [inline Split.xcodeproj/project.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings:1]: ℹ️ Generated by `swift package generate-xcodeproj`

File: Split.xcodeproj/project.xcworkspace/contents.xcworkspacedata [hunk 6]
```diff
@@ -2,6 +2,6 @@
 <Workspace
```
Comment (bsarrazin) [inline Split.xcodeproj/project.xcworkspace/contents.xcworkspacedata:2]: ℹ️ Generated by `swift package generate-xcodeproj`

File: Split.xcodeproj/xcshareddata/xcschemes/Split.xcscheme [hunk 7]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (bsarrazin) [inline Split.xcodeproj/xcshareddata/xcschemes/Split.xcscheme:1]: ℹ️ Generated by `swift package generate-xcodeproj` as `Split-Package.xcscheme`, renamed to `Split.xcscheme`

File: Split/ExportedImports.swift [hunk 8]
```diff
@@ -0,0 +1,3 @@
+#if SWIFT_PACKAGE
```
Comment (bsarrazin) [inline Split/ExportedImports.swift:1]: ℹ️ This directive command is to prevent buidl failures when consuming through CocoaPods

**Review by javrudsky — APPROVED** [review]
LGTM! Thank you @bsarrazin  for your contribution!

**Pull request merged**